core-rust/core-api-server/src/core_api/handlers/lts/state_account_deposit_behaviour.rs

```diff
@@ -0,0 +1,245 @@
+                        empty_virtual_account_resource_specific_bahaviour(resource_address),
+                    )
+                })
+                .collect()
+        }),
+    )
+}
+
+fn empty_virtual_account_resource_specific_bahaviour(
+    resource_address: ResourceAddress,
+) -> models::ResourceSpecificDepositBehaviour {
+    models::ResourceSpecificDepositBehaviour {
+        resource_preference: None,
+        vault_exists: false,
+        is_xrd: resource_address == XRD,
+        deposit_allowed: true,
+    }
+}
+
+fn response(
+    context: &MappingContext,
+    header: &LedgerHeader,
+    default_deposit_rule: &DefaultDepositRule,
+    badge_of_authorized_depositor: Option<bool>,
+    resource_specific_behaviours: Option<
+        IndexMap<String, models::ResourceSpecificDepositBehaviour>,
+    >,
+) -> Result<Json<models::LtsStateAccountDepositBehaviourResponse>, ResponseError<()>> {
+    Ok(models::LtsStateAccountDepositBehaviourResponse {
+        state_version: to_api_state_version(header.state_version)?,
+        ledger_header_summary: Box::new(to_api_ledger_header_summary(context, header)?),
+        default_deposit_rule: match default_deposit_rule {
+            DefaultDepositRule::Accept => models::DefaultDepositRule::Accept,
+            DefaultDepositRule::Reject => models::DefaultDepositRule::Reject,
+            DefaultDepositRule::AllowExisting => models::DefaultDepositRule::AllowExisting,
+        },
+        badge_of_authorized_depositor,
+        resource_specific_behaviours,
+    })
+    .map(Json)
+}
+
+/// Resolves whether the deposit is allowed, based on raw inputs.
+/// See https://docs-babylon.radixdlt.com/main/scrypto/native-blueprints/accounts.html for a
+/// flow chart.
+fn allows_deposit(
+    default_deposit_rule: &DefaultDepositRule,
+    badge_of_authorized_depositor: &Option<bool>,
+    resource_preference: &Option<AccountResourcePreference>,
+    vault_exists: bool,
+    is_xrd: bool,
+) -> bool {
+    match resource_preference {
+        // An explicit resource preference allows it:
+        Some(AccountResourcePreference::Allowed) => true,
+        // An explicit resource preference disallows it, but a present authorized badge can override this:
+        Some(AccountResourcePreference::Disallowed) => badge_of_authorized_depositor == &Some(true),
+        // No explicit resource preference - try the same logic with the default deposit rule:
+        None => {
+            match default_deposit_rule {
+                // The default rule allows it:
+                DefaultDepositRule::Accept => true,
+                // The default rule disallows it, but a present authorized badge can override this:
+                DefaultDepositRule::Reject => badge_of_authorized_depositor == &Some(true),
+                // The extra case of the default rule, allowing existing vaults *or* XRD:
+                DefaultDepositRule::AllowExisting => vault_exists || is_xrd,
```

Comment:
I think this is incorrect. 

Should be `vault_exists || is_xrd || badge_of_authorized_depositor == &Some(true)`?

Maybe @0xOmarA can confirm :-D

Reply:
OMG you are right, and even the flow chart in the official docs agrees 😱 

Basically it looks like presenting a badge of authorized depositor overrides everything, so I now just moved this condition first, PTAL.